build failure

failed to build: 
`cargo +nightly build`  results in 
```
  Compiling nrf-modem v0.6.0 (/Users/robert/src/rust/nrf/nrf-modem)
error[E0308]: mismatched types
    --> /Users/robert/src/rust/nrf/nrf-modem/src/gnss.rs:420:68
     |
420  |                         core::mem::transmute::<[i8; 83], [u8; 83]>(data.assume_init().nmea_str); // Make data be u8
     |                         ------------------------------------------ ^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `[i8; 83]`, found `[u8; 83]`
     |                         |
     |                         arguments to this function are incorrect
     |
     = note: expected array `[i8; 83]`
                found array `[u8; 83]`
note: function defined here
    --> /Users/robert/.rustup/toolchains/nightly-aarch64-apple-darwin/lib/rustlib/src/rust/library/core/src/intrinsics/mod.rs:1917:21
     |
1917 | pub const unsafe fn transmute<Src, Dst>(_src: Src) -> Dst {
     |                     ^^^^^^^^^
```

nightly:

```
% rustup toolchain list
stable-aarch64-apple-darwin (default)
nightly-2023-01-24-aarch64-apple-darwin
nightly-aarch64-apple-darwin
1.83-aarch64-apple-darwin
```

building stable fails with:
```
error[E0554]: `#![feature]` may not be used on the stable release channel
 --> /Users/robert/.cargo/registry/src/index.crates.io-6f17d22bba15001f/embedded-hal-async-0.2.0-alpha.1/src/lib.rs:5:1
  |
5 | #![feature(async_fn_in_trait, impl_trait_projections)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove the attribute
  |
  = help: the feature `impl_trait_projections` has been stable since `1.74.0` and no longer requires an attribute to enable
  = help: the feature `impl_trait_projections` has been stable since `1.74.0` and no longer requires an attribute to enable

```

Targeting `nrf9151`

```
% cargo +nightly -vV
cargo 1.85.0-nightly (652623b77 2024-12-20)
release: 1.85.0-nightly
commit-hash: 652623b779c88fe44afede28bf7f1c9c07812511
commit-date: 2024-12-20
host: aarch64-apple-darwin
libgit2: 1.8.1 (sys:0.19.0 vendored)
libcurl: 8.7.1 (sys:0.4.74+curl-8.9.0 system ssl:(SecureTransport) LibreSSL/3.3.6)
ssl: OpenSSL 1.1.1w  11 Sep 2023
os: Mac OS 14.7.1 [64-bit]
```

yeah - it looks like there's s till a bit of nightly dependency even after removing features in all files:
```
% cargo build
   Compiling embassy-embedded-hal v0.1.0 (https://github.com/embassy-rs/embassy.git?rev=4b30394#4b303949)
error[E0658]: `try` expression is experimental
  --> /Users/robert/.cargo/git/checkouts/embassy-9312dcb0ed774b29/4b30394/embassy-embedded-hal/src/shared_bus/asynch/spi.rs:69:46
   |
69 |           let op_res: Result<(), BUS::Error> = try {
   |  ______________________________________________^
70 | |             for buf in operations {
71 | |                 bus.read(buf).await?;
72 | |             }
73 | |         };
   | |_________^
   |
   = note: see issue #31436 <https://github.com/rust-lang/rust/issues/31436> for more information

```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -17,9 +17,10 @@ jobs:
       - uses: dtolnay/rust-toolchain@master
         with:
           toolchain: stable
-          components: clippy, rustfmt, llvm-tools-preview
+          components: clippy, rustfmt, llvm-tools
           targets: thumbv8m.main-none-eabihf
 
       - run: cargo fmt -- --check
       - run: cargo clippy --features nrf9160 --target thumbv8m.main-none-eabihf
+      - run: cargo clippy --features nrf9160,dns-async,defmt --target thumbv8m.main-none-eabihf
       - run: cargo clippy --features nrf9151 --target thumbv8m.main-none-eabihf

--- a/.vscode/settings.json
+++ b/.vscode/settings.json
@@ -1,0 +1,3 @@
+{
+    "rust-analyzer.cargo.features": ["nrf9160"]
+}

--- a/src/dns/dns_async.rs
+++ b/src/dns/dns_async.rs
@@ -1,8 +1,8 @@
 use crate::{dns::dns_cache::DnsCache, CancellationToken, Error, LteLink, UdpSocket};
+use core::net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr, SocketAddrV4};
 use core::{cell::RefCell, convert::TryInto};
 use embassy_sync::{blocking_mutex::raw::ThreadModeRawMutex, mutex::Mutex};
 use embassy_time::Duration;
-use no_std_net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr, SocketAddrV4};
 
 const DNS_SERVERS: [[u8; 4]; 8] = [
     [8, 8, 8, 8],
@@ -81,7 +81,7 @@ struct DNSQuestion<'a> {
     qclass: u16,
 }
 
-impl<'a> DNSQuestion<'a> {
+impl DNSQuestion<'_> {
     fn write(&self, buffer: &mut [u8], mut pos: usize) -> Result<usize, Error> {
         // Encode the domain name into the QNAME field
         for label in self.qname.split('.') {
@@ -358,7 +358,7 @@ async fn process_dns_response(response: &[u8], transaction_id: u16) -> Result<Ip
                     .lock()
                     .await
                     .borrow_mut()
-                    .insert(&hostname, &addr, ttl)?;
+                    .insert(hostname, &addr, ttl)?;
                 return Ok(addr);
             }
             28 if rdlength == 16 => {
@@ -368,7 +368,7 @@ async fn process_dns_response(response: &[u8], transaction_id: u16) -> Result<Ip
                     .lock()
                     .await
                     .borrow_mut()
-                    .insert(&hostname, &addr, ttl)?;
+                    .insert(hostname, &addr, ttl)?;
                 return Ok(addr);
             }
             _ => {

--- a/src/dns/dns_cache.rs
+++ b/src/dns/dns_cache.rs
@@ -1,5 +1,5 @@
 use crate::Error;
-use no_std_net::{IpAddr, Ipv4Addr, Ipv6Addr};
+use core::net::{IpAddr, Ipv4Addr, Ipv6Addr};
 
 /// The maximum domain name length
 const MAX_DOMAIN_LEN: usize = 256;
@@ -51,13 +51,13 @@ impl DnsCache {
         match ip {
             IpAddr::V4(ipv4_addr) => defmt::trace!(
                 "create dns cache entry with {} -> {} and ttl {}",
-                core::str::from_utf8(&hostname).unwrap(),
+                core::str::from_utf8(hostname).unwrap(),
                 ipv4_addr.octets(),
                 ttl
             ),
             IpAddr::V6(ipv6_addr) => defmt::trace!(
                 "create dns cache entry with {} -> {} and ttl {}",
-                core::str::from_utf8(&hostname).unwrap(),
+                core::str::from_utf8(hostname).unwrap(),
                 ipv6_addr.octets(),
                 ttl
             ),

--- a/src/dtls_socket.rs
+++ b/src/dtls_socket.rs
@@ -149,7 +149,7 @@ pub struct DtlsReceiveSocket<'a> {
     socket: &'a DtlsSocket,
 }
 
-impl<'a> DtlsReceiveSocket<'a> {
+impl DtlsReceiveSocket<'_> {
     fn socket(&self) -> &Socket {
         &self.socket.inner
     }
@@ -161,7 +161,7 @@ pub struct DtlsSendSocket<'a> {
     socket: &'a DtlsSocket,
 }
 
-impl<'a> DtlsSendSocket<'a> {
+impl DtlsSendSocket<'_> {
     fn socket(&self) -> &Socket {
         &self.socket.inner
     }

--- a/src/error.rs
+++ b/src/error.rs
@@ -87,6 +87,20 @@ impl embedded_io_async::Error for Error {
             Error::ModemAlreadyInitialized => embedded_io_async::ErrorKind::Other,
             Error::TlsPacketTooBig => embedded_io_async::ErrorKind::Other,
             Error::NoSecurityTag => embedded_io_async::ErrorKind::Other,
+            #[cfg(feature = "dns-async")]
+            Error::DomainNameTooLong => embedded_io_async::ErrorKind::InvalidInput,
+            #[cfg(feature = "dns-async")]
+            Error::DnsCacheOverflow => embedded_io_async::ErrorKind::Other,
+            #[cfg(feature = "dns-async")]
+            Error::DnsHeaderBufferOverflow => embedded_io_async::ErrorKind::Other,
+            #[cfg(feature = "dns-async")]
+            Error::DnsQuestionBufferOverflow => embedded_io_async::ErrorKind::Other,
+            #[cfg(feature = "dns-async")]
+            Error::DnsSocketTimeout => embedded_io_async::ErrorKind::TimedOut,
+            #[cfg(feature = "dns-async")]
+            Error::DnsSocketError => embedded_io_async::ErrorKind::Other,
+            #[cfg(feature = "dns-async")]
+            Error::DnsParseFailed => embedded_io_async::ErrorKind::Other,
         }
     }
 }

--- a/src/gnss.rs
+++ b/src/gnss.rs
@@ -416,8 +416,9 @@ impl GnssData {
                     )
                     .into_result()?;
 
-                    let data =
-                        core::mem::transmute::<[i8; 83], [u8; 83]>(data.assume_init().nmea_str); // Make data be u8
+                    // Allow transmute warnings. Sometimes apparently the nmea_str is a `u8` array already
+                    #[allow(clippy::useless_transmute, clippy::missing_transmute_annotations)]
+                    let data = core::mem::transmute::<_, [u8; 83]>(data.assume_init().nmea_str); // Make data be u8
                     let mut string_data = ArrayString::from_byte_string(&data)?;
                     string_data.truncate(
                         string_data

--- a/src/tcp_stream.rs
+++ b/src/tcp_stream.rs
@@ -187,7 +187,7 @@ pub struct TcpReadStream<'a> {
     stream: &'a TcpStream,
 }
 
-impl<'a> TcpReadStream<'a> {
+impl TcpReadStream<'_> {
     fn socket(&self) -> &Socket {
         &self.stream.inner
     }
@@ -203,7 +203,7 @@ pub struct TcpWriteStream<'a> {
     stream: &'a TcpStream,
 }
 
-impl<'a> TcpWriteStream<'a> {
+impl TcpWriteStream<'_> {
     fn socket(&self) -> &Socket {
         &self.stream.inner
     }

--- a/src/tls_stream.rs
+++ b/src/tls_stream.rs
@@ -243,7 +243,7 @@ pub struct TlsReadStream<'a> {
     socket: &'a TlsStream,
 }
 
-impl<'a> TlsReadStream<'a> {
+impl TlsReadStream<'_> {
     fn socket(&self) -> &Socket {
         &self.socket.inner
     }
@@ -259,7 +259,7 @@ pub struct TlsWriteStream<'a> {
     socket: &'a TlsStream,
 }
 
-impl<'a> TlsWriteStream<'a> {
+impl TlsWriteStream<'_> {
     fn socket(&self) -> &Socket {
         &self.socket.inner
     }

--- a/src/udp_socket.rs
+++ b/src/udp_socket.rs
@@ -135,7 +135,7 @@ pub struct UdpReceiveSocket<'a> {
     socket: &'a UdpSocket,
 }
 
-impl<'a> UdpReceiveSocket<'a> {
+impl UdpReceiveSocket<'_> {
     fn socket(&self) -> &Socket {
         &self.socket.inner
     }
@@ -148,7 +148,7 @@ pub struct UdpSendSocket<'a> {
     socket: &'a UdpSocket,
 }
 
-impl<'a> UdpSendSocket<'a> {
+impl UdpSendSocket<'_> {
     fn socket(&self) -> &Socket {
         &self.socket.inner
     }

